tools/compiletest/src/runtest.rs

```diff
@@ -387,6 +392,99 @@ impl<'test> TestCx<'test> {
         }
     }
 
+    /// Modify the results section of the output from the new format to the old
+    /// format so that it works for existing `expected` files that are based on
+    /// the old format
+    fn reformat_output(output: &str) -> String {
+        // Process the output on a line-by-line basis
+        let lines: Vec<String> = output.split('\n').map(|s| s.to_string()).collect();
+        let mut adapted_lines: Vec<String> = Vec::new();
+
+        // 1. Copy all lines until and including the "RESULTS:" line, which
+        // marks the beginning of the results section of the output
+        let mut line_iter = lines.iter();
+        while let Some(line) = line_iter.next() {
+            adapted_lines.push(line.clone());
+            if adapted_lines.last().unwrap() == "RESULTS:" {
+                break;
+            }
+        }
+
+        if line_iter.clone().peekable().peek().is_none() {
+            // iterated over all the lines, so this is not the new format
+            return output.to_string();
+        }
+
+        // 2. Adapt output of the form:
+        //
+        //        Check 2: foo.assertion.2
+        //        - Status: SUCCESS
+        //        - Description: "assertion failed: x == 2"
+        //
+        //    to:
+        //
+        //       [foo.assertion.2] line X assertion failed: x == 2: SUCCESS
+        //
+        // (Note that since the line numbers are missing from the new format, it
+        // will be printed as "line X")
+        let mut check_index = 1usize;
+        while let Some(line) = line_iter.next() {
+            // Check line is of the form: "Check <n>: foo.assertion.5"
+            if let Some(name) = line.strip_prefix(format!("Check {}: ", check_index).as_str()) {
+                // Result line is of the form: "\t - Status: <Result>"
+                let result = line_iter.next().unwrap().strip_prefix("\t - Status: ").unwrap();
```

Comment:
Can you call String::trim() to get rid of the whitespace before comparing the string result here and everywhere else? This seems quite fragile.